### DISCUSSION TITLE
idle_delete_ttl on google provider seems to do not work

### DISCUSSION THREAD

**Discussion by bmoisson** [apache/airflow#17465] [Q&A] [answered]
Hello,

I am not able to set the idle_delete_ttl when I use the DataprocCreateClusterOperator from google provider. 

Here is the configuration : 

 `
from airflow.providers.google.cloud.operators.dataproc import (
    DataprocCreateClusterOperator,
    DataprocDeleteClusterOperator,
    DataprocSubmitJobOperator)
from google.protobuf.duration_pb2 import Duration

...

message_duration = Duration()

CLUSTER_CONFIG = {
   ...
    ,
    "lifecycle_config": {
        "idle_delete_ttl": message_duration.FromSeconds(1000)
    }
}

...

 create_dataproc_cluster = DataprocCreateClusterOperator(
        task_id="create_dataproc_cluster",
        project_id=project_id,
        gcp_conn_id=gcp_conn_id,
        cluster_config=CLUSTER_CONFIG,
        region=models.Variable.get('gce_region'),
        cluster_name=CLUSTER_NAME,
    )
`
The cluster is created but the idle do not appear. 

Do I miss something or is it a bug?

**Comment by githubwua** [comment, marked as answer]
This question is also raised somewhere else, and I will answer it here so that other users can also see the answer.

TLDR: Specify 1000 instead of message_duration.FromSeconds(1000) for idle_delete_ttl

Details;
When submitting a Dataproc cluster creation request to the Dataproc API endpoint, the request payload is in JSON format, and the "idle_delete_ttl" field actually expects a number (as a string like "1000s"), not protobuf duration.  While it is true that protobuf duration is indeed used internally in Dataproc as suggested by the [documentation](https://cloud.google.com/dataproc/docs/reference/rest/v1/ClusterConfig#lifecycleconfig), but externally, the API accepts this field in JSON format, and this is why we need to specify a number for this field when using the Dataproc API.

According to [the Dataproc Operator documentation](https://airflow.apache.org/docs/apache-airflow-providers-google/1.0.0/_modules/airflow/providers/google/cloud/operators/dataproc.html#ClusterGenerator._build_lifecycle_config), the field expects an Optional[int]

        idle_delete_ttl: Optional[int] = None,

And the unit is seconds.

**Reply by bmoisson** [reply]
To complete this answer with an answer from google : 

I should write : 

 "idle_delete_ttl": { "seconds": 1000 } to make it work
